include/mqtt/client.hpp

```diff
@@ -7,12 +7,15 @@
 #if !defined(MQTT_CLIENT_HPP)
 #define MQTT_CLIENT_HPP
 
-#include <set>
-#include <memory>
+#include <mqtt/variant.hpp> // should be top to configure variant limit
```

Comment:
Some of boost library (e.g. lexical_cast) sets BOOST_MPL_LIMIT_LIST_SIZE internally. In order to avoid error, I placed `#include <mqtt/variant.hpp>` before any other boost libraries inclusion. The comment means that.

https://wandbox.org/permlink/I56Sjw9cNpVl0D0r